core/src/main/java/io/questdb/griffin/engine/groupby/GroupByLong128HashSet.java

```diff
@@ -187,7 +188,9 @@ private long probe(long lo, long hi, long index) {
             if (loKey == lo && hiKey == hi) {
                 return -index - 1;
             }
-        } while (true);
+        } while (index != index0);
+
+        throw CairoException.critical(0).put("corrupt hash table");
```

Comment:
```suggestion
        throw CairoException.critical(0).put("corrupt long128 hash set state");
```

Reply:
Changed in 20ba60c6b4d95ff2aec27a0f99f028032b891afe